Introduce lab mode for visualizations

ppisljar opened the pull request:
Release Note: Newly introduced visualizations can now be part of labs-mode. Visualizations in labs-mode introduce new more cutting-edge functionality and can be subject to change across minor releases.  Labs-mode can be turned off in the advanced settings. Labs-visualization will then no longer be available to the user. The Time Series Visual Builder is not part of labs-mode, it continues to be an experimental feature. The input controls are the first to be flagged as a lab visualization.

![image](https://user-images.githubusercontent.com/1833023/33785535-76bfabd4-dc33-11e7-94b3-a97e358fd77c.png)


---------------------------
continued from #14907

Release Note: You can now disable visualizations in lab mode via a setting. Visualizations in experimental and lab mode will show an information, that they are still experimental.

timroes commented:
Jenkins, test this

ppisljar commented:
which test failed ?

timroes commented:
"visualize app tile map visualize app tile map chart should zoom in to level 10"

https://kibana-ci.elastic.co/job/elastic+kibana+pull-request+multijob-selenium/8892//console

alexfrancoeur commented:
Screenshots and note below.

Time Series Visual Builder (Experimental)

<img width="295" alt="screen shot 2017-11-20 at 9 56 20 am" src="https://user-images.githubusercontent.com/25181823/33028101-eed2e4a6-cde2-11e7-89d4-f0352a4d5b15.png">

<img width="742" alt="screen shot 2017-11-20 at 9 56 39 am" src="https://user-images.githubusercontent.com/25181823/33028145-09f5acfa-cde3-11e7-8639-c5d775a975f8.png">
<img width="1186" alt="screen shot 2017-11-20 at 10 00 10 am" src="https://user-images.githubusercontent.com/25181823/33028746-b1a9046e-cde4-11e7-8e4d-50d54e47440a.png">

Controls (Labs)

<img width="292" alt="screen shot 2017-11-20 at 9 56 23 am" src="https://user-images.githubusercontent.com/25181823/33028109-f3a6bdea-cde2-11e7-8d1e-06a8f9397f22.png">

<img width="783" alt="screen shot 2017-11-20 at 9 56 31 am" src="https://user-images.githubusercontent.com/25181823/33028130-fdc02a00-cde2-11e7-8cc1-ec6e699d37f4.png">

This seems to be a bit inconsistent. If we're using the `lab` label in one description, we should probably use it in both. Or use `experimental` in both. I know we briefly discussed this, but would it make sense to repurpose this sentence so labs/experimental warnings are pluggable or directly reflected by the "state" variable?

This is `an experimental` visualization. Have feedback? Please create an issue in GitHub.
This is `a lab` visualization. Have feedback? Please create an issue in GitHub.

The GitHub link didn't actually seem to be working

<img width="1680" alt="screen shot 2017-11-20 at 9 58 27 am" src="https://user-images.githubusercontent.com/25181823/33028419-e9b9a90e-cde3-11e7-8237-0e8ad89aa3c3.png">

The advanced setting is called `visualize:enableLabsVisualizations`. Using "Labs" makes sense to reference the overall feature, however when we describing a visualization that is part of "Labs" it should be called a "Lab Visualization". The current name of the setting is a bit odd to me. Could we use `visualize:enableLabs`? The description is self explanatory.

<img width="1186" alt="screen shot 2017-11-20 at 10 00 10 am" src="https://user-images.githubusercontent.com/25181823/33028443-fccdd524-cde3-11e7-9dbd-2c367f9dec2e.png">

This issue still seems to be there for deleting a visualization. Can we make sure this doesn't get checked into 6.1?

<img width="995" alt="screen shot 2017-11-20 at 11 03 49 am" src="https://user-images.githubusercontent.com/25181823/33028494-18d0b0ac-cde4-11e7-8c16-ef1795c77712.png">

ppisljar commented:
`fail: "visualize app tile map visualize app tile map chart should zoom in to level 10"`

seems that test is flaky as well :(

ppisljar commented:
- changed labs to lab in the text
- changed setting name to visualize:enableLabs
- fixed github link

not sure what you mean with issue when deleting ? (that box looks the same on master, no text inside just warning in the title)

lets keep the experimental/labs text as it is now, we can easily change that later in a fix as we would be changing labels only.

ppisljar commented:
jenkins, test this

timroes commented:
Jenkins, test this

ppisljar commented:
`"visualize app tile map visualize app tile map chart should zoom in to level 10"`

jenkins, test this

ppisljar commented:
backport: 
6.x: d68bc2a
